js/src/dapps/registry/accounts/actions.js

```diff
@@ -0,0 +1,23 @@
+const { personal } = window.parity.api;
+
+export const set = (accounts) => ({ type: 'accounts set', accounts });
+
+export const fetch = () => (dispatch) => {
```

Comment:
What you mention sounds a lot like [`redux-actions`](https://github.com/acdlite/redux-actions), which a @jacogr in fact uses in parity.

I agree that every failure & success should be represented in form of an action. Also, the UI should always handle both.

Wether the `redux-actions` model is the right one and how to name the actions is debatable however. I think we will have to discuss Redux further.